skutil/preprocessing/tests/test_impute.py

```diff
@@ -144,6 +144,7 @@ def test_selective_imputer():
     assert_fails(SelectiveImputer(def_fill=['a', 'b', 'c']).fit, TypeError, a)
     assert_fails(SelectiveImputer(def_fill='a').fit, TypeError, a)
     assert_fails(SelectiveImputer(def_fill=[1, 2, 'a']).fit, TypeError, a)
+    assert_fails(SelectiveImputer().fit, TypeError, a)
```

Comment:
So I believe that the value of `def_fill` is passed to `_BaseImputer` via `SelectiveImputer`'s `__init__`:

```python
def __init__(self, cols=None, as_df=True, def_fill='mean'):
        super(SelectiveImputer, self).__init__(cols, as_df, def_fill)
```

in `_BaseImputer` when `def_fill` is set it sets the value of `self._fill` in its `__init__` : 

```python
def __init__(self, cols=None, as_df=True, def_fill=None):
        self.cols = cols
        self.as_df = as_df
        self.fill_ = self._def_fill if def_fill is None else def_fill
```

Ultimately [this is the line of code](https://github.com/tgsmith61591/skutil/blob/develop/skutil/preprocessing/impute.py#L226-L229) I am trying to test. When `None` is passed to `def_fill` and I think that is why it is not failing properly because the code overrides it with `_def_fill = -999999` and it doesn't hit the final block. 

I think in order to test it we want the value of def_fill to be:
- not iterable
- not a string type
- not numeric

So an appropriate datatype would have to be passed into def_fill to satisfy those requirements.